### DISCUSSION TITLE
html rendered as text and not html

### DISCUSSION THREAD

**Discussion by mmailaender** [ueberdosis/tiptap#5090] [Tiptap Editor]
I'm trying to set HTML to the content of the editor, but it seems that TipTap is converting the HTML to their entities (`<p>Tom</p>` => `&lt;p&gt;Tom&lt;/p&gt;`) leading to the problem, that the HTML syntax will be visualized instead of interpreted
![image](https://github.com/ueberdosis/tiptap/assets/87228994/577b5ed7-a212-45b3-8649-2838d49efb83)

```js
console.log(editor.getHTML()); // <p></p>
console.log(defaultValue); // <p>Tom</p>

editor.commands.setContent(defaultValue)

console.log(editor.getHTML()); // <p>&lt;p&gt;Tom&lt;/p&gt;</p>
```

Is there a way to enforce HTML? In the TipTap examples, it works out of the box, but for us, this is not the case.

**Comment by Sadashii** [comment]
> I'm trying to set HTML to the content of the editor, but it seems that TipTap is converting the HTML to their entities (`<p>Tom</p>` => `&lt;p&gt;Tom&lt;/p&gt;`) leading to the problem, that the HTML syntax will be visualized instead of interpreted ![image](https://private-user-images.githubusercontent.com/87228994/324468293-577b5ed7-a212-45b3-8649-2838d49efb83.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzUzMjE1MzIsIm5iZiI6MTc3NTMyMTIzMiwicGF0aCI6Ii84NzIyODk5NC8zMjQ0NjgyOTMtNTc3YjVlZDctYTIxMi00NWIzLTg2NDktMjgzOGQ0OWVmYjgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDA0VDE2NDcxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwMDU0NmJmNGNmMzJmZDc4Y2JkNjQ1M2M2ZTVlMTg5YmYyMTFmMjc3ZGFlYTljNzJhN2QyNTdiMTQ3MGVjNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2IYI2IRr8Ee1MGOBf0zp2w0lMfWvh5eJsuOjYox_TOc)
> 
> ```js
> console.log(editor.getHTML()); // <p></p>
> console.log(defaultValue); // <p>Tom</p>
> 
> editor.commands.setContent(defaultValue)
> 
> console.log(editor.getHTML()); // <p>&lt;p&gt;Tom&lt;/p&gt;</p>
> ```
> 
> Is there a way to enforce HTML? In the TipTap examples, it works out of the box, but for us, this is not the case.

This usually happens when Tiptap's parser decides your input should be treated as literal plain text (or Markdown) rather than raw HTML. 

In a vanilla Tiptap installation, `editor.commands.setContent('<p>Tom</p>')` works out of the box and parses as HTML. Since it's escaping the tags into `&lt;p&gt;` in your environment, something in your specific configuration is intercepting it. 

Here are the three most common culprits and how to fix them:

### 1. Are you using `tiptap-markdown`?
If you have the `tiptap-markdown` extension installed, it overrides Tiptap's default `setContent` behavior to expect Markdown strings instead of HTML. By default, it will escape arbitrary HTML tags to prevent XSS. 

**The Fix:** You need to explicitly tell the Markdown extension to parse HTML.
```javascript
Markdown.configure({
  html: true, // This stops it from escaping HTML into entities
})

**Comment by arnaugomez** [comment]
Does this issue also happen when you set the `contentType` parameter to `'html'`?

```ts
editor.commands.setContent('<p>Tom</p>', { contentType: 'html' })
```
